framework/src/main/java/org/checkerframework/framework/util/Contract.java

```diff
@@ -136,17 +140,37 @@ private Contract(
      * @param contractAnnotation the pre- or post-condition annotation that the programmer wrote;
      *     used for diagnostic messages
      * @param ensuresQualifierIf the ensuresQualifierIf field, for a conditional postcondition
+     * @param atypeFactory used for standardizing annotations
+     * @param context used for standardizing annotations
+     * @param pathToMethodDecl used for standardizing annotations
      * @return a new contract
      */
-    public static Contract create(
+    protected static Contract create(
             Kind kind,
             String expressionString,
             AnnotationMirror annotation,
             AnnotationMirror contractAnnotation,
-            Boolean ensuresQualifierIf) {
+            Boolean ensuresQualifierIf,
+            GenericAnnotatedTypeFactory<?, ?, ?, ?> atypeFactory,
+            JavaExpressionContext context,
+            TreePath pathToMethodDecl) {
         if ((ensuresQualifierIf != null) != (kind == Kind.CONDITIONALPOSTCONDITION)) {
             throw new BugInCF("Mismatch: ensuresQualifierIf=%s, kind=%s", ensuresQualifierIf, kind);
         }
+
+        // pathToMethodDecl is null if the method is not declared in source code.
+        // TODO: The annotations still need to be standardized in that case.  We don't currently
```

Comment:
Just a note: This is not standardizing the contract annotation.  `annotation` is the qualifier the expression in the contract should be.

Reply:
Because `annotation` is a type qualifier, it is standardized elsewhere.  You should not also standardize it here.  If you delete this code, the test case I added passes. You can also delete the `atMethodSignature` boolean to `org.checkerframework.framework.util.dependenttypes.DependentTypesHelper#standardizeString`.  That code was working around the fact that this was already standardizing the annotation.

Reply:
> Just a note: This is not standardizing the contract annotation. `annotation` is the qualifier the expression in the contract should be.

Right.  That is in the Javadoc for this method (`Contract.create`).

Reply:
See https://github.com/mernst/checker-framework/pull/11 which removes this code.